### PR TITLE
[FrameworkBundle] Exclude unreadable files when executing About command

### PR DESCRIPTION
**Pull request by michaljusiega**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix https://github.com/symfony/symfony/issues/39916
| License       | MIT
| Doc PR        | ~

Fix as explained in: https://github.com/symfony/symfony/issues/39916.

Tests failed looks like not releated to my changes.

### CODE REVIEW

**Review by ro0NL — APPROVED** [review]
works for me

File: src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        $this->assertStringContainsString('Log directory', $tester->getDisplay());
+
+        $this->fs->chmod($this->kernel->getCacheDir().'/readable_file', 0777);
+    }
+
+    public function testAboutWithUnreadableFiles()
+    {
+        // skip test on Windows; PHP can't easily set file as unreadable on Windows
+        if ('\\' === \DIRECTORY_SEPARATOR) {
+            $this->markTestSkipped('This test cannot run on Windows.');
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php:62]: Why? We are using `is_readable` a lot in the code, I would be interested in understanding the issues on Windows.

File: src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    public function testAboutWithReadableFiles()
+    {
+        $this->fs->dumpFile($this->kernel->getCacheDir().'/readable_file', str_repeat('XXX', 1000000));
+        $this->fs->chmod($this->kernel->getCacheDir().'/readable_file', 0777);
+
+        $tester = $this->createCommandTester();
+        $ret = $tester->execute([]);
+
+        $this->assertEquals(0, $ret, 'Returns 0 in case of success');
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php]: The third parameter can be removed it does not bring any value (same for the other similar parameters).

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php (hunk 1 above)
Comment (michaljusiega) [reply src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php:62]: I created this test based on `FilesystemTest`, see https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/Filesystem/Tests/FilesystemTest.php#L44

File: src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php (hunk 2 above)
Comment (nicolas-grekas) [reply src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php]: please use assertSame also

Comment (michaljusiega) [reply src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php]: Done

**Review by michaljusiega — COMMENTED** [review]
Ready

File: src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    {
+        try {
+            $this->fs->remove($this->kernel->getProjectDir());
+        } catch (IOException $e) {
+        }
+    }
+
+    public function testAboutWithReadableFiles()
+    {
+        $this->fs->dumpFile($this->kernel->getCacheDir().'/readable_file', str_repeat('XXX', 1000000));
```
Comment (javiereguiluz) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php]: If the actual file contents are not important, we could slightly improve test performance if we just do something like this:

```php
$this->fs->dumpFile($this->kernel->getCacheDir().'/readable_file', 'The file content.');
```

Comment (michaljusiega) [reply src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php]: Done

File: src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+{
+    /** @var TestAppKernel */
+    private $kernel;
+    /** @var Filesystem */
+    private $fs;
+
+    protected function setUp(): void
+    {
+        $this->fs = new Filesystem();
+        $this->kernel = new TestAppKernel('test', true);
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php]: You should create these 2 vars in each test as here, you could leak memory (the alternative would be to unset those in teardown, but I think it's cleaner to just do everything in each test method as the duplicate code is simple enough).

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php (hunk 4 above)
Comment (michaljusiega) [reply src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php]: Done.

**Comment by michaljusiega** [comment]
@fabpot Can We finish this PR ?

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @michaljusiega.

**Pull request merged**